src/seidr.py

```diff
@@ -44,49 +75,36 @@ def plot(self):
 
         fig = plt.figure(facecolor='w')
         ax = fig.add_subplot(111, facecolor='#dddddd', axisbelow=True)
-        ax.plot(self.t, S/1000, 'b', alpha=0.5, lw=2, label='Susceptible')
-        ax.plot(self.t, E/1000, 'orange', alpha=0.5, lw=2, label='Exposed')
-        ax.plot(self.t, I/1000, 'r', alpha=0.5, lw=2, label='Infected')
-        ax.plot(self.t, D/1000, 'black', alpha=0.5, lw=2, label='Fatalities')
-        ax.plot(self.t, R/1000, 'g', alpha=0.5, lw=2, label='Recovered with immunity')
+        ax.plot(self.t, S, 'b', alpha=0.5, lw=2, label='Susceptible')
+        ax.plot(self.t, E, 'orange', alpha=0.5, lw=2, label='Exposed')
+        ax.plot(self.t, I, 'r', alpha=0.5, lw=2, label='Infected')
+        ax.plot(self.t, D, 'black', alpha=0.5, lw=2, label='Fatalities')
+        ax.plot(self.t, R, 'g', alpha=0.5, lw=2, label='Recovered with immunity')
         ax.set_xlabel('Time /days')
-        ax.set_ylabel('Number (1000s)')
-        ax.set_ylim(0,1.2)
+        ax.set_ylabel('Population')
         ax.yaxis.set_tick_params(length=0)
         ax.xaxis.set_tick_params(length=0)
         ax.grid(visible=True, which='major', c='w', lw=2, ls='-')
         legend = ax.legend()
         legend.get_frame().set_alpha(0.5)
         for spine in ('top', 'right', 'bottom', 'left'):
             ax.spines[spine].set_visible(False)
-        plt.show()
-
+        plt.savefig(f"{OUTPUT_DIRECTORY_PATH}/SEIR.png")
+        plt.close()
 
-if __name__ == "__main__":
-    # Total population, N.
-    N = 1000
-    # Initial number of infected and recovered individuals, I0 and R0.
-    I0, E0, D0, R0 = 0, 100, 0, 0
-    # Everyone else, S0, is susceptible to infection initially.
-    S0 = N - I0 - R0 - D0 - E0
-    # Contact rate, beta, and mean recovery rate, gamma, (in 1/days).
-    beta, gamma = 0.5, 1./10 
-    # A grid of time points (in days)
-    # print(t)
-    mi = 0.0002  # natural death rate
-    alpha = 0.01  # fatality rate
-    lambda_ = 0.0003*N  # natural birth rate
-    epsilon = 0.2  # exposed to infectious
-    # Initial conditions vector
-    y0 = S0, E0, I0, D0, R0
-    t = 160  # number of days to compute
-
-    model = SEIDR( N, beta, gamma, mi, lambda_, epsilon, alpha, t)
-    
-    model.calculate(y0)
+    def save_results(self):
+        if self.results is None:
+            raise ModelResultsNotCalculatedError()
 
-    model.plot()
+        S, E, I, D, R = self.results.T
+        results = zip( S, E, I, D, R)
+        df = pd.DataFrame(results, columns=["Susceptible", "Exposed", "Infected", "Fatalities", "Recovered"])
+        df.to_csv(f"{OUTPUT_DIRECTORY_PATH}/results.csv")
```

Comment:
maybe let's parametrize the filename with the model parameters somehow, in case we want to compute data for two different models and compare?